src/index.js

```diff
@@ -151,13 +151,14 @@ class Plugin {
 	getLogMetricCF(alarm, functionName, normalizedFunctionName){
 		var output = {};
 		if (alarm.pattern) {
-			const logMetricCFRefBase = this.naming.getLogMetricCFRef(normalizedFunctionName,alarm.name);
+			const stackName = _.upperFirst(_.camelCase(this.serverless.getProvider('aws').naming.getStackName()));
+			const logMetricCFRefBase = this.naming.getLogMetricCFRef(normalizedFunctionName,stackName,alarm.name);
 			const logMetricCFRefALERT = `${logMetricCFRefBase}ALERT`;
 			const logMetricCFRefOK = `${logMetricCFRefBase}OK`;
 			const CFLogName = this.serverless.getProvider('aws').naming.getLogGroupLogicalId(functionName);
-			const metricNamespace = `${this.serverless.service.service}_${alarm.namespace}`;
+			const metricNamespace = alarm.namespace;
 			const logGroupName =  this.serverless.getProvider('aws').naming.getLogGroupName(this.serverless.service.getFunction(functionName).name);
-			const metricName = `${alarm.metric}${normalizedFunctionName}`;
+			const metricName = `${stackName}${alarm.metric}${normalizedFunctionName}`;
```

Comment:
The same metric name is created in two places. Feels like it should be moved to a separate function inside `Naming` class. The same applies to stack name, but I'm not sure if it should be a method in `Naming` or just in this file, as `Naming` class does not know anything about `serverless`, which IMHO is not right.